twitchio/http.py

```diff
@@ -40,12 +40,41 @@ class HTTPSession:
 
     def __init__(self, loop, **attrs):
         self.client_id = client_id = attrs.get('client_id', None)
+        self.client_secret = attrs.get("client_secret", None)
+        self.token = attrs.get("api_token", None)
+        self.scopes = attrs.get("scopes", [])
 
         if not client_id:
-            log.warning('Running without client ID, some HTTP endpoints may not work without authentication.')
+            log.warning('Running without client ID, HTTP endpoints will not work without authentication.')
+
+        if not self.token and not client_id:
+            log.warning("Running without client ID or bearer token, HTTP endpoints will not work without authentication.")
 
         self._bucket = RateBucket(method='http')
         self._session = aiohttp.ClientSession(loop=loop)
+        self._refresh_token = None
+
+    async def generate_token(self):
+        if not self.client_id or not self.client_secret:
+            raise HTTPException("unable to generate a token")
+
+        if self._refresh_token:
+            url = "https://id.twitch.tv/oauth2/token?grant_type=refresh_token&refresh_token={0}&client_id={1}&client_secret={2}".format(
+                self._refresh_token, self.client_id, self.client_secret)
+
```

Comment:
Maybe make `https://id.twitch.tv/oauth2/token` a constant outside of this scope.